### ISSUE TITLE
Android

### ISSUE THREAD

**Issue by iakshay** [rishabhp/ama#12]
You have plenty of posts on android on your blog. 

As a full stack developer who has worked with both frontend and backend technologies, how has your experience been on android? What do you like/dislike about the two platforms and communities. 

Also, can you share details about the android app you are developing?

**Comment by rishabhp** [comment]
Apps have been gaining immense traction for the past couple of years, mobile is the new hot thing which is why I got interested in app development, specifically Android because of its predominance in India. I've only about half a year of experience in native Android app development, hence my comparison of web vs android might not be highly apt.

I believe it's hard to find good Android resources to learn and most of them are of low quality. The documentation is decent but at times hard for new comers to wrap their head around various concepts (probably because they're written by senior experienced android devs who already have a context set in their mind while writing). Developing UIs just feels very constricted compared to web development. Most of the UI components are libraries that are built-in which cannot be customised to a great extent. You've to pass configuration values to the component objects to modify the look/feel/behaviour. Building custom components requires good amount of experience and skill. Compared to this, the web is really simple and it's super easy to build custom components/widgets. I guess all this is because of the age of both the platforms.

Community-wise it can be hard to seek Android help at times. Some questions might go un-answered on various forums. Some problems (UI related) are probably too complicated to be solved and you've to resort to simple existing solutions (or even WebViews).

That said, I've moved out from native Android development since a long time. I'd love to continue but now I feel like react native might be the solution to easily build native mobile apps with the power of web technologies (that has awesome debugging tools).

There are 2 incomplete apps I worked on:

- [Pyc](https://github.com/rishabhp/Pyc) - Idea was basically a person can share a collection of photos with his contacts just by tagging them. This would end up creating a separate communication group where the sharer and tagged people would be able to swipe through the photos and chat on them individually.
- [Loop](https://github.com/rishabhp/Loop) - Tinder-like app but instead of swiping photos you'd get connected to people around you over a 10-20 seconds video call (webrtc, websockets) and then swipe to like/dislike.
